### ISSUE TITLE
What do I need to install on Ubuntu to meet the "bullet" dependency?

### ISSUE THREAD

**Issue by pscollins** [Revolutionary-Games/Thrive#368]
I've grabbed everything off of apt-get that has "libbullet" in the name, but cmake isn't happy:

    CMake Error at /usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:108 (message):
      Could NOT find Bullet (missing: BULLET_DYNAMICS_LIBRARY_DEBUG
      BULLET_COLLISION_LIBRARY_DEBUG BULLET_MATH_LIBRARY_DEBUG
      BULLET_SOFTBODY_LIBRARY_DEBUG)
    Call Stack (most recent call first):
      /usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:315     (_FPHSA_FAILURE_MESSAGE)
      cmake_modules/FindBullet.cmake:74 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
      CMakeLists.txt:124 (find_package)

Do I need to [build it from source](https://github.com/bulletphysics/bullet3)? I'm a little unclear on the status of this project re: Linux since I see that getting a Linux build working is on your list for the next release, but I can't find any open issues or threads on the dev forums regarding what the issue with Linux is. A list of packages to `apt-get install` in the README would make it much easier to get started, assuming that it can be built on Linux at the moment.

**Comment by hhyyrylainen** [comment]
Are you trying to configure Thrive in Debug configuration?
The packages are probably missing bullet libs compiled in Debug configuration, so if you want to use those you have to compile from source. 
But I recommend using RelWithDebInfo which compiles fine on my machine, I haven't tried to get Debug configuration to work so I don't know if that even can be made to work.
So you could try running cmake like this: `cmake .. -DCMAKE_BUILD_TYPE=RelWithDebInfo`

**Comment by pscollins** [comment]
I'm just trying to run `cmake .` in the root directory. Here's the full error message:

    ➜  Thrive git:(master) ✗ cmake . -DCMAKE_BUILD_TYPE=RelWithDebInfo 
    -- Looking for OGRE...
    -- Found Ogre Byatis (1.8.1)
    -- Found OGRE: optimized;/usr/lib/x86_64-linux-gnu/libOgreMain.so;debug;/usr/lib/x86_64-linux-gnu/libOgreMain.so
    -- Looking for OGRE_Paging...
    -- Found OGRE_Paging: optimized;/usr/lib/x86_64-linux-gnu/libOgrePaging.so;debug;/usr/lib/x86_64-linux-gnu/libOgrePaging.so
    -- Looking for OGRE_Terrain...
    -- Found OGRE_Terrain: optimized;/usr/lib/x86_64-linux-gnu/libOgreTerrain.so;debug;/usr/lib/x86_64-linux-gnu/libOgreTerrain.so
    -- Looking for OGRE_Property...
    -- Found OGRE_Property: optimized;/usr/lib/x86_64-linux-gnu/libOgreProperty.so;debug;/usr/lib/x86_64-linux-gnu/libOgreProperty.so
    -- Looking for OGRE_RTShaderSystem...
    -- Found OGRE_RTShaderSystem: optimized;/usr/lib/x86_64-linux-gnu/libOgreRTShaderSystem.so;debug;/usr/lib/x86_64-linux-gnu/libOgreRTShaderSystem.so
    CMake Error at cmake_modules/FindBoost.cmake:1192 (message):
      Unable to find the requested Boost libraries.
    
      Boost version: 1.54.0
    
      Boost include path: /usr/include
    
      The following Boost libraries could not be found:
    
              boost_filesystem
    
      Some (but not all) of the required Boost libraries were found.  You may
      need to install these additional Boost libraries.  Alternatively, set
      BOOST_LIBRARYDIR to the directory containing Boost libraries or BOOST_ROOT
      to the location of Boost.
    Call Stack (most recent call first):
      CMakeLists.txt:105 (find_package)
    
    
    CMake Error at /usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:108 (message):
      Could NOT find Bullet (missing: BULLET_DYNAMICS_LIBRARY_DEBUG
      BULLET_COLLISION_LIBRARY_DEBUG BULLET_MATH_LIBRARY_DEBUG
      BULLET_SOFTBODY_LIBRARY_DEBUG)
    Call Stack (most recent call first):
      /usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:315 (_FPHSA_FAILURE_MESSAGE)
      cmake_modules/FindBullet.cmake:74 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
      CMakeLists.txt:124 (find_package)
    
    
    -- Configuring incomplete, errors occurred!
    
Can the project be built on Linux at the moment?

**Comment by Moopli** [comment]
It can be built completely, but we've been getting a persistent runtime segfault. So far we've made a bit of progress fixing it, we can at least replicate it, and hopefully with more eyes, and time, on it, we can fix it.

But that's runtime, so, yes, it can at least be built. I don't think I did anything special to Bullet, you can take a look [at this branch](https://github.com/Revolutionary-Games/Thrive/compare/linux-compatches) to see what I have.

**Comment by pscollins** [comment]
Do you remember how you got bullet installed? Did you have to build from source? It would be much easier if I could just use apt-get.

Is there a bug report or something that summarizes the existing problem on Linux? 

(It's worth noting that I don't know C++ very well --- I have more experience with C --- so I doubt I'll be able to contribute too much, but it would be interesting to poke at)

Even after trying to install from source, I wasn't able to get it to find bullet, so I swapped out the dependencies in FindBullet.cmake on `*_DEBUG` targets for their non-debug-flavored variants and that quieted that down, but now it's complaining about CEGUI:

    CMake Error at /usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:108 (message):
      Could NOT find CEGUI (missing: CEGUI_LIBRARY)
    Call Stack (most recent call first):
      /usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:315 (_FPHSA_FAILURE_MESSAGE)
      cmake_modules/FindCEGUI.cmake:181 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
      CMakeLists.txt:153 (find_package)

**Comment by Moopli** [comment]
According to dpkg I installed bullet through apt-get:
```
$ dpkg -l | grep bullet
ii  libbullet-dev:amd64                                         2.81-rev2613+dfsg2-1                       amd64        professional 3D Game Multiphysics Library -- development files
ii  libbullet-extras-dev:amd64                                  2.81-rev2613+dfsg2-1                       amd64        professional 3D Game Multiphysics Library -- extra development
ii  libbullet2.81-dbg:amd64                                     2.81-rev2613+dfsg2-1                       amd64        professional 3D Game Multiphysics Library -- debugging symbols
ii  libbulletcollision2.81:amd64                                2.81-rev2613+dfsg2-1                       amd64        professional 3D Game Multiphysics Library -- collision
ii  libbulletdynamics2.81:amd64                                 2.81-rev2613+dfsg2-1                       amd64        professional 3D Game Multiphysics Library -- dynamics
ii  libbulletfileloader2.81:amd64                               2.81-rev2613+dfsg2-1                       amd64        professional 3D Game Multiphysics Library -- file loader
ii  libbulletsoftbody2.81:amd64                                 2.81-rev2613+dfsg2-1                       amd64        professional 3D Game Multiphysics Library -- soft body
ii  libbulletworldimporter2.81:amd64                            2.81-rev2613+dfsg2-1                       amd64        professional 3D Game Multiphysics Library -- world importer
ii  libbulletxmlworldimporter2.81:amd64                         2.81-rev2613+dfsg2-1                       amd64        professional 3D Game Multiphysics Library -- xml world importer
```

I added `/usr/lib/x86_64-linux-gnu/` to `CEGUI_LIBRARY_SEARCH_DIR` in `FindCEGUI.cmake`, as CEGUI's install directory, it seems, had changed.

**Comment by Moopli** [comment]
oops

**Comment by pscollins** [comment]
I had all of those packages through apt-get but no dice. I've checked out your branch so that change was already made for me; I've installed everything matching `libcegui*` through apt-get and it doesn't like it. Oh well, maybe I'll just wait until you all get the linux situation worked out and check back in a few months.

**Comment by hhyyrylainen** [comment]
My information might be outdated, but when I set up Thrive to work with Ogre 2.0 it required CEGUI to be built from the default branch. It seems unlikely that Ogre 2.0 support would be shipped in a stable CEGUI build anytime soon.I might create an install script for Linux once I start working on Thrive again. 

-------- Alkuperäinen viesti --------
Lähettäjä: Patrick Collins <notifications@github.com> 
Päivämäärä: 09.12.2015  3.58  (GMT+02:00) 
Saaja: Revolutionary-Games/Thrive <Thrive@noreply.github.com> 
Kopio: Henri Hyyryläinen <henri.hyyrylainen@gmail.com> 
Aihe: Re: [Thrive] What do I need to install on Ubuntu to meet the "bullet"
  dependency? (#368) 

I had all of those packages through apt-get but no dice. I've checked out your branch so that change was already made for me; I've installed everything matching libcegui* through apt-get and it doesn't like it. Oh well, maybe I'll just wait until you all get the linux situation worked out and check back in a few months.

**Comment by Moopli** [comment]
Ah hold on, for CEGUI we build from source, default branch, I'm on `cfce8d308878`.

**Comment by jjonj** [comment]
Sounds good @pscollins, sorry that it's not in working condition atm!

@hhyyrylainen We made a ubuntu 14.04 vmware virtual machine and set up thrive on it and as moopli said, replicated the moopli is having issue. I have a download link for the full VM here: https://mega.nz/#!xMoGVSRY!X1atUbzqiBdhob0ITlEEGZblosxqR8uIVcr2IBcNw0A 
in case you have some time and is curious what the issue looks like. (VM is 3gb download, ~12 gb unpacked)

**Comment by hhyyrylainen** [comment]
I had to change Thrive CMakeLists.txt to not try to use ${BULLET_DYNAMICS_LIBRARY_DEBUG} on Linux to make it start compiling. Now I just need to fix some compile errors and I can test whether I get a working executable.

**Comment by jjonj** [comment]
Sounds good! Yeah the bullet error is a bit tricky as we can only build bullet in debug mode on windows, we'll have to find a way to get it to work on both platforms!  
When I built in the virtual machine I took notes, so if you run into other problems there might be a solution in this document: https://docs.google.com/document/d/1FGMzODYC8fRpW9uTvpN7YJ8vPJitA3GBuo998efgxJk/edit especially under "jjonjs vm commands:
"
There was some issues with bullet, cAudio, cegui, ogre ffmpeg player and openal you might run into as well to which we found solutions.
